Source/com/drew/metadata/eps/EpsReader.java

```diff
@@ -0,0 +1,466 @@
+package com.drew.metadata.eps;
+
+import com.drew.imaging.tiff.TiffProcessingException;
+import com.drew.imaging.tiff.TiffReader;
+import com.drew.lang.*;
+import com.drew.lang.annotations.NotNull;
+import com.drew.metadata.Directory;
+import com.drew.metadata.Metadata;
+import com.drew.metadata.exif.ExifTiffHandler;
+import com.drew.metadata.icc.IccReader;
+import com.drew.metadata.photoshop.PhotoshopReader;
+import com.drew.metadata.xmp.XmpReader;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Reads file passed in through SequentialReader and parses encountered data:
+ *      - Basic EPS Comments
+ *      - Photoshop
+ *      - ICC Profile
+ *      - XMP
+ * EPS comments are retrieved from EPS directory.  Photoshop, ICC Profile, and XMP processing
+ * is passed to their respective reader.
+ *
+ * EPS Constraints (Source: https://www-cdf.fnal.gov/offline/PostScript/5001.PDF pg.18):
+ *      - Max line length is 255 characters
+ *      - Lines end with a CR(0xD) or LF(0xA) character
+ *      - ':' separates keywords (considered part of the keyword)
+ *      - Whitespace is either a space(0x20) or tab(0x9)
+ *      - If there is more than one header, the 1st is truth
+ *
+ * @author Payton Garland
+ */
+public class EpsReader {
+
+    private int pos = 0;
+    private RandomAccessStreamReader reader;
+    private int previousTag;
+
+    /**
+     * Filter method that determines if file will contain an EPS Header.  If it does, it will read the necessary
+     * data and then set the position to the beginning of the PostScript data.  If it does not, the position will not
+     * be changed.  After both scenarios, the main extract method is called.
+     *
+     * @param reader RandomAccessStreamReader to read bytes of file
+     * @param metadata Metadata to add directory to and extracted data
+     *
+     */
+    public void extract(@NotNull final RandomAccessStreamReader reader, @NotNull final Metadata metadata) throws IOException
+    {
+        this.reader = reader;
+        EpsDirectory directory = new EpsDirectory();
+        metadata.addDirectory(directory);
+        this.pos = 0;
+        this.reader = reader;
+
+        byte[] magicNumber = reader.getBytes(pos, 4);
```

Comment:
Could be nice to do this without assigning a `byte[]`. If you read out a 32-bit integer, no heap assignment occurs, and the comparison below becomes simpler.